sync/class.jetpack-sync-settings.php

```diff
@@ -53,12 +54,28 @@ static function get_setting( $setting ) {
 			return self::$settings_cache[ $setting ];
 		}
 
-		$value = get_option( self::SETTINGS_OPTION_PREFIX . $setting );
+		if ( self::is_network_setting( $setting ) ) {
+			if ( is_multisite() ) {
+				$value = get_site_option( self::SETTINGS_OPTION_PREFIX . $setting );
+			} else {
+				// On single sites just return the default setting
+				$value = Jetpack_Sync_Defaults::get_default_setting( $setting );
+				self::$settings_cache[ $setting ] = $value;
+				return $value;
+			}
+		} else {
+			$value = get_option( self::SETTINGS_OPTION_PREFIX . $setting );
+		}
 
-		if ( false === $value ) {
-			$default_name = "default_$setting"; // e.g. default_dequeue_max_bytes
-			$value        = Jetpack_Sync_Defaults::$$default_name;
-			update_option( self::SETTINGS_OPTION_PREFIX . $setting, $value, true );
+		if ( false === $value ) { // no default value is set.
+			// We set one so that it gets autoloaded
+			$value        = Jetpack_Sync_Defaults::get_default_setting( $setting );
+			if ( substr( $setting, 0, strlen('network_') ) === 'network_' ) {
```

Comment:
Seems like we could use `self::is_network_setting()` here.

```suggestion
			if ( self::is_network_setting( $setting ) ) {
```